modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/AmqpConnectionManager.java

```diff
@@ -111,17 +119,17 @@ public void forceDisconnectChannel(int connectionId, int channelId, String reaso
      *
      * @return a list of {@link AmqpChannelView} representing AMQP channels
      */
-    public Collection<AmqpChannelView> getChannelViews(int connectionId) {
-        AmqpConnectionHandler connectionHandler;
-        connectionHandler = connectionHandlers.get(connectionId);
+    public Collection<AmqpChannelView> getChannelViews(int connectionId) throws ResourceNotFoundException {
+        AmqpConnectionHandler connectionHandler = getConnectionOrThrow(connectionId);
+        return connectionHandler.getChannelViews();
+    }
+
+    private AmqpConnectionHandler getConnectionOrThrow(int connectionId) throws ResourceNotFoundException {
```

Comment:
In a way, this gives you an idea of what happens if we ask for the wrong connection :)

Since this is a private method do we need to be that explicit? We haven't named users of this methods as `getChannelViewsOrThrow(` etc. WDYT?